src/SignalR/server/Core/src/DefaultHubLifetimeManager.cs

```diff
@@ -341,7 +343,18 @@ public override async Task<T> InvokeConnectionAsync<T>(string connectionId, stri
         }
 
         var invocationId = Interlocked.Increment(ref _lastInvocationId).ToString(NumberFormatInfo.InvariantInfo);
-        using var _ = CancellationTokenUtils.CreateLinkedToken(cancellationToken,
+
+        CancellationTokenSource? cts = null;
+        // Add a default timeout if one isn't provided
+        if (!cancellationToken.CanBeCanceled)
+        {
+            cts = _cancellationTokenSourcePool.Rent();
+            cts.CancelAfter(TimeSpan.FromSeconds(30));
```

Comment:
What stops this timer?

Reply:
The dispose via `using` on line 356 first calls `TryReset` which will cleanup the timer, if that fails it will dispose the cts.

Reply:
Ok. Consider adding a comment for other people reading the source.